src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/InvalidUrlRewriteFormatException.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Xml;
+using System.Xml.Linq;
+
+namespace Microsoft.AspNetCore.Rewrite.Internal.IISUrlRewrite
+{
+    public class InvalidUrlRewriteFormatException : FormatException
+    {
+        public InvalidUrlRewriteFormatException(XElement element, string message)
+            : base(FormatMessage(element, message))
+        {
+        }
+
+        public InvalidUrlRewriteFormatException(XElement element, string message, Exception innerException)
+            : base(FormatMessage(element, message), innerException)
+        {
+        }
+
+        private static string FormatMessage(XElement element, string message)
+        {
+            var lineInfo = (IXmlLineInfo)element;
+            var line = lineInfo.LineNumber;
+            var col = lineInfo.LinePosition;
```

Comment:
If you're going to have a custom exception type then you may as well add these fields as properties so they're progamatically accessible.